process/block/shardblock.go

```diff
@@ -232,6 +226,16 @@ func (sp *shardProcessor) ProcessBlock(
 		}
 	}()
 
+	processedMetaHdrs, err := sp.getProcessedMetaBlocks(body, header.MetaBlockHashes)
```

Comment:
use function getProcessedMetablocksFromHeader, it gives you all the metablocks which are fully processed.

Reply:
processMetablocksFromHeader also does some setting of flags which is not needed when only processing, and this function is used as well in createBlock where there is no header available.

Reply:
common logic of the 2 functions should be moved together. making a base function, and one which sets the processed miniblocks as well. Even if that means iterating 2 times over the same set of data, it is only metablocks, which is at most a few elements.

Reply:
done.